.golangci.yml

```diff
@@ -0,0 +1,64 @@
+run:
+  timeout: 5m
+  tests: false
+  skip-dirs:
+    - client/v2/common
+  
+linters:
+  disable-all: true
+  enable:
+    - deadcode
+    - errcheck
+    - gci
+    - gofmt
+    - gosimple
+    - govet
+    - ineffassign
+    - misspell
+    - revive
+    - staticcheck
+    - structcheck
+    - typecheck
+    - unused
+    - varcheck
```

Comment:
`go-algorand` also includes the following:
```yml
linters:
...
    - exportloopref
    - nilerr
    - nolintlint
    - paralleltest
```

Chat GPT explains:
* **exportloopref**: This linter checks for pointers to enclosing loop variables. These can lead to subtle bugs due to how Go handles variable scoping and closures.
* **nilerr**: This linter checks that you are not ignoring errors returned by functions. Ignoring errors can lead to unexpected behavior.
* **nolintlint**: This linter checks for incorrect or unnecessary //nolint comments. The //nolint comment is used to tell golangci-lint to ignore certain issues on a line of code or a whole file.
* **paralleltest**: This linter checks that you are correctly using the t.Parallel() function in your tests to allow them to run in parallel.

Do you think these are worth adding?

Reply:
Good suggestion, I'm in favor of adding the first three linters. I think `paralleltest` is more intrusive and caused more discussion in go-algorand. It's mostly to speed up tests by forcing parallelization, but since tests run pretty quickly in this repo, I think it would be better left out. 

Also thanks for the ChatGPT summary - saved me some clicks 😀